test/base.py

```diff
@@ -73,6 +73,9 @@ def get_properties(self, original: bool = True):
         return_value["start_date"] = self.start_date
         return return_value
 
+    def to_map(self, raw_metadata):
+        return {tuple(md['breadcrumb']): md['metadata'] for md in raw_metadata}
```

Comment:
Why not use the `singer-python` version?
https://github.com/singer-io/singer-python/blob/0c066de21111d8572425083b4a8792d193c80af1/singer/metadata.py#L4-L5

Reply:
Ah, I see this was done with regards to the double logging